2016-08-01 09:57:59,865:INFO:Pegasus.tools.properties(235): processing properties file /scratch/advanced_bioconductor_workflow/submit/samrat/pegasus/advanced_bioconductor_workflow/run0003/pegasus.7652148965149861770.properties... 

2016-08-01T09:57:59-0700 - pegasus-monitord starting - pid 25879 ---------------

Created Pegasus database in: sqlite:////scratch/advanced_bioconductor_workflow/submit/samrat/pegasus/advanced_bioconductor_workflow/run0003/advanced_bioconductor_workflow-0.stampede.db
1470070683 - 2016-08-01T09:58:03-0700 - MONITORD_STARTED  - 26ce8d2d-fe57-41d9-ac0c-a6ad771a6274 - advanced_bioconductor_workflow-0
2016-08-01 10:11:36,491:ERROR:Pegasus.db.modules.stampede_loader.Analyzer(371): (IntegrityError) columns lfn_id, pfn, site are not unique u'INSERT INTO rc_pfn (lfn_id, pfn, site) VALUES (?, ?, ?)' (12, 'file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData', 'local')
Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/stampede_loader.py", line 369, in hard_flush
    self.session.commit()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/scoping.py", line 149, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 719, in commit
    self.transaction.commit()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 352, in commit
    self._prepare_impl()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 332, in _prepare_impl
    self.session.flush()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1814, in flush
    self._flush(objects)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1896, in _flush
    flush_context.execute()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/unitofwork.py", line 372, in execute
    rec.execute(self)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/unitofwork.py", line 525, in execute
    uow
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/persistence.py", line 63, in save_obj
    table, insert)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/persistence.py", line 565, in _emit_insert_statements
    execute(statement, params)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 664, in execute
    params)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 764, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 878, in _execute_context
    context)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 871, in _execute_context
    context)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/default.py", line 320, in do_execute
    cursor.execute(statement, parameters)
IntegrityError: (IntegrityError) columns lfn_id, pfn, site are not unique u'INSERT INTO rc_pfn (lfn_id, pfn, site) VALUES (?, ?, ?)' (12, 'file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData', 'local')
2016-08-01 10:11:36,828:ERROR:Pegasus.db.modules.stampede_loader.Analyzer(372): Integrity error on batch flush: batch will need to be committed per-event which will take longer
2016-08-01 10:11:39,142:ERROR:Pegasus.db.modules.stampede_loader.Analyzer(426): (IntegrityError) columns lfn_id, pfn, site are not unique u'INSERT INTO rc_pfn (lfn_id, pfn, site) VALUES (?, ?, ?)' (12, 'file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData', 'local')
Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/stampede_loader.py", line 423, in individual_commit
    event.commit_to_db(self.session)
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/schema.py", line 113, in commit_to_db
    self._commit(session, batch)
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/schema.py", line 103, in _commit
    session.flush()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/scoping.py", line 149, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1814, in flush
    self._flush(objects)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1896, in _flush
    flush_context.execute()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/unitofwork.py", line 372, in execute
    rec.execute(self)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/unitofwork.py", line 525, in execute
    uow
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/persistence.py", line 63, in save_obj
    table, insert)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/persistence.py", line 565, in _emit_insert_statements
    execute(statement, params)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 664, in execute
    params)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 764, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 878, in _execute_context
    context)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 871, in _execute_context
    context)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/default.py", line 320, in do_execute
    cursor.execute(statement, parameters)
IntegrityError: (IntegrityError) columns lfn_id, pfn, site are not unique u'INSERT INTO rc_pfn (lfn_id, pfn, site) VALUES (?, ?, ?)' (12, 'file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData', 'local')
2016-08-01 10:11:39,158:ERROR:Pegasus.db.modules.stampede_loader.Analyzer(427): Insert failed for event <class 'Pegasus.db.schema.RCPFN'>:
  * wf_id : 1
  * wf_uuid : 26ce8d2d-fe57-41d9-ac0c-a6ad771a6274
  * pfn_id : None
  * lfn : lowAbundance-sce-1.RData
  * site : local
  * pfn : file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData
  * lfn_id : 12
  * event : stampede.rc.pfn

2016-08-01 10:11:40,196:WARNING:Pegasus.monitoring.workflow(104): NL-LOAD-ERROR --> 26ce8d2d-fe57-41d9-ac0c-a6ad771a6274 - advanced_bioconductor_workflow-0
2016-08-01 10:11:40,207:WARNING:Pegasus.monitoring.workflow(105): error sending event: job_inst.main.term --> {'status': 0, 'xwf__id': '26ce8d2d-fe57-41d9-ac0c-a6ad771a6274', 'js__id': 3, 'job_inst__id': 34, 'job__id': 'e.normalize.2_ID0000012', 'ts': 1470071495, 'sched__id': '170559.0'}
2016-08-01 10:11:40,260:WARNING:Pegasus.monitoring.workflow(106): Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/Pegasus/monitoring/workflow.py", line 99, in output_to_db
    self._sink.send(event, kwargs)
  File "/usr/lib64/python2.6/site-packages/Pegasus/monitoring/event_output.py", line 190, in send
    self._db.notify(d)
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/__init__.py", line 146, in notify
    raise ProcessException(str(err))
ProcessException: Instance <JobInstance at 0x3045950> is not present in this Session

2016-08-01 10:11:40,261:ERROR:Pegasus.monitoring.workflow(107): Disabling database population for workflow 26ce8d2d-fe57-41d9-ac0c-a6ad771a6274 in directory /scratch/advanced_bioconductor_workflow/submit/samrat/pegasus/advanced_bioconductor_workflow/run0003
1470071850 - 2016-08-01T10:17:30-0700 - MONITORD_FINISHED - 26ce8d2d-fe57-41d9-ac0c-a6ad771a6274 - advanced_bioconductor_workflow-0

2016-08-01T10:17:30-0700 - pegasus-monitord ending - pid 25879 -----------------

Final Command line options are ['-N', 'advanced_bioconductor_workflow-0.dag.dagman.out'] 
2016-08-01T10:17:31-0700 - pegasus-monitord - DB flushing beginning ------------
2016-08-01 10:17:31,390:ERROR:Pegasus.db.modules.stampede_loader.Analyzer(371): (IntegrityError) columns lfn_id, pfn, site are not unique u'INSERT INTO rc_pfn (lfn_id, pfn, site) VALUES (?, ?, ?)' (12, 'file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData', 'local')
Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/stampede_loader.py", line 369, in hard_flush
    self.session.commit()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/scoping.py", line 149, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 719, in commit
    self.transaction.commit()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 352, in commit
    self._prepare_impl()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 332, in _prepare_impl
    self.session.flush()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1814, in flush
    self._flush(objects)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1896, in _flush
    flush_context.execute()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/unitofwork.py", line 372, in execute
    rec.execute(self)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/unitofwork.py", line 525, in execute
    uow
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/persistence.py", line 63, in save_obj
    table, insert)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/persistence.py", line 565, in _emit_insert_statements
    execute(statement, params)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 664, in execute
    params)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 764, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 878, in _execute_context
    context)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 871, in _execute_context
    context)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/default.py", line 320, in do_execute
    cursor.execute(statement, parameters)
IntegrityError: (IntegrityError) columns lfn_id, pfn, site are not unique u'INSERT INTO rc_pfn (lfn_id, pfn, site) VALUES (?, ?, ?)' (12, 'file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData', 'local')
2016-08-01 10:17:31,391:ERROR:Pegasus.db.modules.stampede_loader.Analyzer(372): Integrity error on batch flush: batch will need to be committed per-event which will take longer
2016-08-01 10:17:31,397:ERROR:Pegasus.db.modules.stampede_loader.Analyzer(426): (IntegrityError) columns lfn_id, pfn, site are not unique u'INSERT INTO rc_pfn (lfn_id, pfn, site) VALUES (?, ?, ?)' (12, 'file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData', 'local')
Traceback (most recent call last):
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/stampede_loader.py", line 423, in individual_commit
    event.commit_to_db(self.session)
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/schema.py", line 113, in commit_to_db
    self._commit(session, batch)
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/schema.py", line 103, in _commit
    session.flush()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/scoping.py", line 149, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1814, in flush
    self._flush(objects)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1896, in _flush
    flush_context.execute()
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/unitofwork.py", line 372, in execute
    rec.execute(self)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/unitofwork.py", line 525, in execute
    uow
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/persistence.py", line 63, in save_obj
    table, insert)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/persistence.py", line 565, in _emit_insert_statements
    execute(statement, params)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 664, in execute
    params)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 764, in _execute_clauseelement
    compiled_sql, distilled_params
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 878, in _execute_context
    context)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/base.py", line 871, in _execute_context
    context)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/engine/default.py", line 320, in do_execute
    cursor.execute(statement, parameters)
IntegrityError: (IntegrityError) columns lfn_id, pfn, site are not unique u'INSERT INTO rc_pfn (lfn_id, pfn, site) VALUES (?, ?, ?)' (12, 'file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData', 'local')
2016-08-01 10:17:31,398:ERROR:Pegasus.db.modules.stampede_loader.Analyzer(427): Insert failed for event <class 'Pegasus.db.schema.RCPFN'>:
  * wf_id : 1
  * wf_uuid : 26ce8d2d-fe57-41d9-ac0c-a6ad771a6274
  * pfn_id : None
  * lfn : lowAbundance-sce-1.RData
  * site : local
  * pfn : file:///scratch/pegasus-scratch/samrat/pegasus/advanced_bioconductor_workflow/run0003/lowAbundance-sce-1.RData
  * lfn_id : 12
  * event : stampede.rc.pfn

2016-08-01 10:17:31,498:WARNING:pegasus-monitord(216): could not call the finish method in the nl loader class... exiting anyway
2016-08-01 10:17:31,539:WARNING:pegasus-monitord(217): Traceback (most recent call last):
  File "/usr/bin/pegasus-monitord", line 213, in finish_stampede_loader
    sink.close()
  File "/usr/lib64/python2.6/site-packages/Pegasus/monitoring/event_output.py", line 196, in close
    self._db.finish()
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/stampede_loader.py", line 1165, in finish
    self.hard_flush()
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/stampede_loader.py", line 374, in hard_flush
    self.hard_flush(batch_flush=False, retry=retry)
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/stampede_loader.py", line 366, in hard_flush
    self.individual_commit(event, merge=True)
  File "/usr/lib64/python2.6/site-packages/Pegasus/db/modules/stampede_loader.py", line 424, in individual_commit
    self.session.expunge(event)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/scoping.py", line 149, in do
    return getattr(self.registry(), name)(*args, **kwargs)
  File "/usr/lib64/pegasus/externals/python/sqlalchemy/orm/session.py", line 1292, in expunge
    orm_util.state_str(state))
InvalidRequestError: Instance <JobInstance at 0x3045950> is not present in this Session

2016-08-01T10:17:31-0700 - pegasus-monitord - DB flushing ended ----------------
2016-08-01T10:17:31-0700 - pegasus-monitord - DB flushing beginning ------------
2016-08-01T10:17:31-0700 - pegasus-monitord - DB flushing ended ----------------
